mst-algorithms/mst-algorithms/main.cpp

```diff
@@ -4,42 +4,33 @@
 //
 //  Created by Yinon Horev on 13/04/2022.
 //
-using namespace std;
 
 #include <iostream>
 #include "graph.hpp"
+#include "UnionFind.hpp"
+using namespace std;
 
 
-void display_AdjList(adjNode* ptr, int i);
 
 int main(int argc, const char * argv[]) {
-    // graph edges array.
-    graphEdge edges[] = {
-        // (x, y, w) -> edge from x to y with weight w
-        {0,1,2},{0,2,4},{1,4,3},{2,3,2},{3,1,4},{4,3,3}
-    };
-    int N = 6;      // Number of vertices in the graph
-    // calculate number of edges
-    int n = sizeof(edges)/sizeof(edges[0]);
-    // construct graph
-    DiaGraph diagraph(edges, n, N);
-    // print adjacency list representation of graph
-    cout<<"Graph adjacency list "<<endl<<"(start_vertex, end_vertex, weight):"<<endl;
-    for (int i = 0; i < N; i++)
-    {
-        // display adjacent vertices of vertex i
-        display_AdjList(diagraph.head[i], i);
-    }
+    int n = 7;
+    UnionFind test(n);
+    for (int i=0; i <= n; i++) {
+          test.MakeSet(i);
+      }
+      
+    test.Union(0, 1);
+    test.PrintNode(0);
+    test.PrintNode(1);
+    test.Union(1, 2);
+    test.PrintNode(0);
+    test.PrintNode(1);
+    test.PrintNode(2);
+    test.Union(2, 3);
+    test.PrintForest();
+    cout << test.Find(3) << endl;
+    test.PrintForest();
```

Comment:
I think it is best we will implement methods that contains each Class test's, and by that accumulate tests until we submit the assigment.